src/Cli/dotnet/Parser.cs

```diff
@@ -109,6 +109,39 @@ public static Command GetBuiltInCommand(string commandName)
                 .FirstOrDefault(c => c.Name.Equals(commandName, StringComparison.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Implements token-per-line response file handling for the CLI. We use this instead of the built-in S.CL handling
+        /// to ensure backwards-compatibility with MSBuild.
+        /// </summary>
+        public static bool TokenPerLine(string tokenToReplace, out IReadOnlyList<string> replacementTokens, out string errorMessage) {
```

Comment:
this is the core change that gets us back to parity with the response file parsing from the older S.CL version.  We get a presumed file path, and if that file path isn't an actual path error out to the user (S.CL did this previously). Then, we read the lines, strip out data from any # tokens onward, and drop all subsequently-empty lines.